internal/controller/ai_gateway_route.go

```
@@ -311,11 +312,23 @@ func (c *AIGatewayRouteController) reconcileExtProcConfigMap(ctx context.Context
 					if awsCred := backendSecurityPolicy.Spec.AWSCredentials; awsCred.CredentialsFile != nil || awsCred.OIDCExchangeToken != nil {
 						ec.Rules[i].Backends[j].Auth = &filterapi.BackendAuth{
 							AWSAuth: &filterapi.AWSAuth{
-								CredentialFileName: path.Join(backendSecurityMountPath(volumeName), "/credentials"),
+								CredentialFileName: path.Join(backendSecurityMountPath(volumeName), constants.AwsCredentialsKey),
 								Region:             backendSecurityPolicy.Spec.AWSCredentials.Region,
 							},
 						}
 					}
+				case aigv1a1.BackendSecurityPolicyTypeAzureCredentials:
+					if backendSecurityPolicy.Spec.AzureCredentials == nil {
+						return fmt.Errorf("AzureCredentials type selected but not defined %s", backendSecurityPolicy.Name)
+					}
+					if cred := backendSecurityPolicy.Spec.AzureCredentials; cred.ClientSecretRef != nil {
```

Could we return an error if clientSecretRef isn't specified?

ie

```
if cred := backendSecurityPolicy.Spec.AzureCredentials; cred.ClientSecretRef == nil {
  return fmt.Errorf(...)
}

ec.Rules[i]... = ..
```